nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -87,6 +88,12 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
         # self.setup_optimizer_param_groups()
 
         self.megatron_amp_o2 = cfg.get('megatron_amp_O2', False)
+        self.with_distributed_adam = cfg.optim.get('name') == 'distributed_fused_adam'
+
+        if self.megatron_amp_o2 and self.with_distributed_adam:
+            raise ValueError(
+                "Distributed optimizers are not compatible with O2. Please set megatron_amp_O2 to False in the model config."
```

Comment:
@timmoon10 Hi, @erhoo82 mentioned that the when `with_distributed_adam=True` it includes `megatron_amp_o2`. Is that the case?
If it is, it might not be good to say they are not compatible, but O2 is encapsulated in distributed optimizer. Set magatron_amp_O2 to False will make some confusions to users.

Reply:
That's a good point, increasingly I'm coming to the opinion that `megatron_amp_o2` should be required for the distributed optimizer.

Reply:
I like this a lot. This will means can we continue to recommend `megatron_amp_o2` as we do now.